src/core/qgstiles.cpp

```diff
@@ -35,6 +38,38 @@ QgsTileMatrix QgsTileMatrix::fromWebMercator( int zoomLevel )
   return tm;
 }
 
+QgsTileMatrix QgsTileMatrix::fromCustomDef( int zoomLevel, const QgsCoordinateReferenceSystem &crs,
+                                            const QgsPointXY &z0TopLeftPoint, const double &z0Dimension,
+                                            const int &z0MatrixWidth, const int &z0MatrixHeight )
+{
+  // Constant for square extent calculation
+  const double tileSize = 256.0;
+  const double POINTS_TO_M = 2.800005600011068 / 10000.0;
```

Comment:
Could you please refer me to the source for this conversion? I am a bit confused - initially I thought that was "pt" (as in typographical points) to meters, but in fact it must be "pixels" to meters at some DPI?

Reply:
Isn't scale calculation meant to use "standardized rendering pixel size" where 1px = 0.28mm as defined in WMS and WMTS standards?

With that value even for web mercator the scale denominator is:
```(20037508.3427892*2 / 256.) / 0.00028 = 559082264.0287166```
which is pretty much the same as the one listed in the code currently (559082264.0287178)

Reply:
I use the standardized rendering pixel size as defined in WMS and WFS, and I found `559081145.864214` instead of `559082264.0287178`

Reply:
Where did you find `559081145.864214` ?

Reply:
From what I understand how web mercator root tile is defined:
- longitude goes from -180 to +180
- latitude goes from -85.0511 to +85.0511  ... the exact value is defined as `arctan(sinh(π))`

Based on that, projected coordinates of the root tile are +/- 20037508.3427892 for both X and Y:
```
>>> ct=QgsCoordinateTransform(QgsCoordinateReferenceSystem("EPSG:4326"),QgsCoordinateReferenceSystem("EPSG:3857"), QgsCoordinateTransformContext())
>>> ct.transform(-180, -85.05112877980659)
<QgsPointXY: POINT(-20037508.34278924390673637 -20037508.34278923645615578)>
>>> ct.transform(180, 85.05112877980659)
<QgsPointXY: POINT(20037508.34278924390673637 20037508.34278924390673637)>
```

And based on that, scale denominator should be `(20037508.34278924390673637 * 2 / 256) / (0.28 / 1000) = 559082264.0287176`

Reply:
I find `559081145.864214` in the test with `2.800005600011068 / 10000.0`

I can simplify the `POINTS_TO_M` constant.